include/wlr/util/addon.h

```diff
@@ -0,0 +1,38 @@
+#ifndef WLR_UTIL_ADDON_H
+#define WLR_UTIL_ADDON_H
+
+#include <wayland-server-core.h>
+
+struct wlr_addon_set {
+	// private state
+	struct wl_list addons;
+};
+
+struct wlr_addon;
+
+struct wlr_addon_interface {
+	const char *name;
+	void (*destroy)(struct wlr_addon *addon);
+};
+
+struct wlr_addon {
+	const struct wlr_addon_interface *impl;
+	void *owner;
```

Comment:
Maybe this could be made `const`, so that the caller can set `owner` to a pointer to a `struct wlr_addon_interface`. And maybe make this private state?

Reply:
Done, although I don't think `owner` should be set to impl. wlroots doesn't have anything global so there always will be something to set as owner anyway.